### PR TITLE
Cleanup PR 

### PR DESCRIPTION
**Pull request by HazarathKumarM**
resolved Doxygen warnings
Node augmentations cleanup

### CODE REVIEW

File: amd_openvx_extensions/amd_opencv/include/vx_ext_opencv.h [hunk 1]
```diff
@@ -83,6 +98,10 @@ extern "C"
 	 * \param [in] ddepth The input <tt>\ref VX_TYPE_INT32</tt> scalar to set output image depth.
 	 * \param [in] kwidth The input <tt>\ref VX_TYPE_UINT32</tt> scalar to set box filter width.
 	 * \param [in] kheight The input <tt>\ref VX_TYPE_UINT32</tt> scalar to set box filter height.
+	 * \param [in] Anchor_X The input <tt>\ref VX_TYPE_UINT32</tt> scalar to set anchor point x.
+	 * \param [in] Anchor_Y The input <tt>\ref VX_TYPE_UINT32</tt> scalar to set anchor point y.
+	 * \param [in] Normalized The input <tt>\ref VX_TYPE_BOOL</tt> scalar for Normalized argument.
+	 * \param [in] Bordertype The input <tt>\ref VX_TYPE_UINT32</tt> scalar to set borderType.
```
Comment (swetha097) [inline amd_openvx_extensions/amd_opencv/include/vx_ext_opencv.h:104]: Should the args be in CAPS ? The rest of the arguments seem to be all in small letters. Please check and confirm on this

File: rocAL/rocAL/include/api/rocal_api.h [hunk 2]
```diff
@@ -33,28 +33,34 @@ THE SOFTWARE.
...
+ * \param [in] batch_size batch size of the rocal context
+ * \param [in] affinity affinity for the rocal context
+ * \param [in] gpu_id GPU id associated with rocal context
+ * \param [in] cpu_thread_count number of cpu threads
+ * \param [in] prefetch_queue_depth The depth of the prefetch queue for the RocalContext (default is 3).
+ * \param [in] output_tensor_data_type The output tensor data type (default is ROCAL_FP32).
+ * \return
+ */
+extern "C" RocalContext ROCAL_API_CALL rocalCreate(size_t batch_size, RocalProcessMode affinity, int gpu_id = 0, size_t cpu_thread_count = 1, size_t prefetch_queue_depth = 3, RocalTensorOutputType output_tensor_data_type = RocalTensorOutputType::ROCAL_FP32);
+// extern "C"  RocalContext  ROCAL_API_CALL rocalCreate(size_t batch_size, RocalProcessMode affinity, int gpu_id = 0, size_t cpu_thread_count = 1);
```
Comment (swetha097) [inline rocAL/rocAL/include/api/rocal_api.h:46]: Can we remove this commented out code ?

File: rocAL/rocAL/include/api/rocal_api_augmentation.h [hunk 3]
```diff
@@ -25,34 +25,28 @@ THE SOFTWARE.
...
+ * \return
  */
 extern "C" RocalTensor ROCAL_API_CALL rocalSequenceRearrange(RocalContext p_context, RocalTensor p_input,
-                                                              std::vector<unsigned int>& new_order,
-                                                              bool is_output);
+                                                             std::vector<unsigned int> &new_order,
+                                                             bool is_output);
 
-/*! \brief Resize images.
+/*! \brief Accepts U8 and RGB24 input.
```
Comment (swetha097) [inline rocAL/rocAL/include/api/rocal_api_augmentation.h:41]: I think this briefing should be retained as Resize images.
And this to be added in \note as above augmentation

File: rocAL/rocAL/include/api/rocal_api_augmentation.h [hunk 4]
```diff
@@ -454,30 +442,30 @@ extern "C" RocalTensor ROCAL_API_CALL rocalFishEye(RocalContext context, RocalTe
                                                    RocalTensorLayout output_layout = ROCAL_NONE,
                                                    RocalTensorOutputType output_datatype = ROCAL_UINT8);
 
-/*! \brief Applies vignette effect on images.
+/*! \brief Accepts U8 and RGB24 input.
```
Comment (swetha097) [inline rocAL/rocAL/include/api/rocal_api_augmentation.h:445]: I think this briefing should be retained and 
Accepts U8 and RGB24 input should be added as part of the /note

File: rocAL/rocAL/include/api/rocal_api_augmentation.h [hunk 5]
```diff
@@ -454,30 +442,30 @@ extern "C" RocalTensor ROCAL_API_CALL rocalFishEye(RocalContext context, RocalTe
...
- * \return RocalTensor
+ * \return
  */
 extern "C" RocalTensor ROCAL_API_CALL rocalVignette(RocalContext context, RocalTensor input,
                                                     bool is_output, RocalFloatParam sdev = NULL,
                                                     RocalTensorLayout output_layout = ROCAL_NONE,
                                                     RocalTensorOutputType output_datatype = ROCAL_UINT8);
 
-/*! \brief Applies vignette effect on images with fixed parameters.
+/*! \brief Accepts U8 and RGB24 input.
```
Comment (swetha097) [inline rocAL/rocAL/include/api/rocal_api_augmentation.h:460]: Retain the brief.
And add this as a part of the Note

File: rocAL/rocAL/include/api/rocal_api_augmentation.h [hunk 6]
```diff
@@ -454,30 +442,30 @@ extern "C" RocalTensor ROCAL_API_CALL rocalFishEye(RocalContext context, RocalTe
...
  * \ingroup group_rocal_augmentations
  * \param [in] context Rocal context
  * \param [in] input Input Rocal tensor
- * \param [in] is_output is the output tensor part of the graph output
  * \param [in] sdev standard deviation for the vignette effect
+ * \param [in] is_output is the output tensor part of the graph output
  * \param [in] output_layout the layout of the output tensor
  * \param [in] output_datatype the data type of the output tensor
- * \return RocalTensor
+ * \return
```
Comment (swetha097) [inline rocAL/rocAL/include/api/rocal_api_augmentation.h:468]: Why was this removed ?
 * \return RocalTensor

File: rocAL/rocAL/include/api/rocal_api_data_loaders.h [hunk 7]
```diff
@@ -145,232 +147,237 @@ extern "C"  RocalTensor  ROCAL_API_CALL rocalJpegCOCOFileSource(RocalContext con
...
+                                                                     bool is_output,
+                                                                     std::vector<float>& area_factor,
+                                                                     std::vector<float>& aspect_ratio,
+                                                                     unsigned num_attempts,
+                                                                     bool shuffle = false,
+                                                                     bool loop = false,
+                                                                     RocalImageSizeEvaluationPolicy decode_size_policy = ROCAL_USE_MOST_FREQUENT_SIZE,
+                                                                     unsigned max_width = 0, unsigned max_height = 0);
+
+/*! Creates JPEG image reader and partial decoder. It allocates the resources and objects required to read and decode COCO Jpeg images stored on the file systems. It has internal sharding capability to load/decode in parallel is user wants. If images are not Jpeg compressed they will be ignored.
```
Comment (swetha097) [inline rocAL/rocAL/include/api/rocal_api_data_loaders.h:181]: Add /*! \brief

File: rocAL/rocAL/include/api/rocal_api_data_loaders.h [hunk 8]
```diff
@@ -145,232 +147,237 @@ extern "C"  RocalTensor  ROCAL_API_CALL rocalJpegCOCOFileSource(RocalContext con
...
+                                                                                unsigned shard_id,
+                                                                                unsigned shard_count,
+                                                                                bool is_output,
+                                                                                std::vector<float>& area_factor,
+                                                                                std::vector<float>& aspect_ratio,
+                                                                                unsigned num_attempts,
+                                                                                bool shuffle = false,
+                                                                                bool loop = false,
+                                                                                RocalImageSizeEvaluationPolicy decode_size_policy = ROCAL_USE_MOST_FREQUENT_SIZE,
+                                                                                unsigned max_width = 0, unsigned max_height = 0);
```
Comment (swetha097) [inline rocAL/rocAL/include/api/rocal_api_data_loaders.h:209]: Brief to be added -

File: rocAL/rocAL/include/api/rocal_api_data_loaders.h [hunk 9]
```diff
@@ -380,372 +387,385 @@ extern "C"  RocalTensor  ROCAL_API_CALL rocalMXNetRecordSourceSingleShard(RocalC
...
+                                                                      unsigned resize_shorter = 0,
+                                                                      unsigned resize_longer = 0,
+                                                                      RocalResizeInterpolationType interpolation_type = ROCAL_LINEAR_INTERPOLATION);
+
+/*! \brief Creates CIFAR10 raw data reader and loader. It allocates the resources and objects required to read raw data stored on the file systems.
+ * \param [in] context Rocal context
+ * \param [in] source_path A NULL terminated char string pointing to the location on the disk
+ * \param [in] rocal_color_format The color format the images will be decoded to.
+ * \param [in] is_output Determines if the user wants the loaded images to be part of the output or not.
+ * \param [in] out_width ; output width
```
Comment (swetha097) [inline rocAL/rocAL/include/api/rocal_api_data_loaders.h:694]: To remove the ";"

File: rocAL/rocAL/include/api/rocal_api_data_transfer.h [hunk 10]
```diff
@@ -24,21 +24,51 @@ THE SOFTWARE.
...
+ * \param [in] tensor_output_type the output type of the tensor data
+ * \param [in] multiplier0 the multiplier for channel 0
+ * \param [in] multiplier1 the multiplier for channel 1
+ * \param [in] multiplier2 the multiplier for channel 2
+ * \param [in] offset0 the offset for channel 0
+ * \param [in] offset1 the offset for channel 1
+ * \param [in] offset2 the offset for channel 2
+ * \param [in] reverse_channels flag to reverse the channel orders
+ * \param [in] output_mem_type the memory type of output tensor buffer
+ * @return Rocal status indicating success or failure
```
Comment (swetha097) [inline rocAL/rocAL/include/api/rocal_api_data_transfer.h:50]: change @return to \return

File: rocAL/rocAL/include/api/rocal_api_info.h [hunk 11]
```diff
@@ -24,66 +24,89 @@ THE SOFTWARE.
 #define MIVISIONX_ROCAL_API_INFO_H
 #include "rocal_api_types.h"
 
-///
-/// \param image
-/// \return The width of the ROCAL's output image in pixels
+/*!
+ * \brief Retrieves the width of the output from the RocalContext.
```
Comment (swetha097) [inline rocAL/rocAL/include/api/rocal_api_info.h:28]: To change the brief to:
Retrieves the width of the output.

File: rocAL/rocAL/include/api/rocal_api_info.h [hunk 12]
```diff
@@ -24,66 +24,89 @@ THE SOFTWARE.
...
+ * \param [in] rocal_context The RocalContext
+ * \return The width of the output.
+ */
 extern "C" int ROCAL_API_CALL rocalGetOutputWidth(RocalContext rocal_context);
 
-///
-/// \param image
-/// \return The height of the ROCAL's output image in pixels. It includes all images in the batch.
+/*!
+ * \brief Retrieves the height of the output from the RocalContext.
```
Comment (swetha097) [inline rocAL/rocAL/include/api/rocal_api_info.h:35]: TO change the brief to Retrieves the height of the output.

File: rocAL/rocAL/include/api/rocal_api_info.h [hunk 13]
```diff
@@ -24,66 +24,89 @@ THE SOFTWARE.
...
+ * \param [in] rocal_context The RocalContext
+ * \return The height of the output.
+ */
 extern "C" int ROCAL_API_CALL rocalGetOutputHeight(RocalContext rocal_context);
 
-///
-/// \param rocal_context
-/// \return The color format of the ROCAL's output. It's equivalent of what's passed to the loaders as input color format.
+/*!
+ * \brief Retrieves the color format of the output from the RocalContext.
```
Comment (swetha097) [inline rocAL/rocAL/include/api/rocal_api_info.h:42]: To check and remove  `from the RocalContext` from the entire file

File: rocAL/rocAL/include/api/rocal_api_meta_data.h [hunk 14]
```diff
@@ -23,185 +23,224 @@ THE SOFTWARE.
...
+/*! \brief get bounding box label
+ * \param [in] rocal_context rocal context
+ * \param [out] buf The user's buffer that will be filled with bounding box label info for the images in the output batch. It needs to be of size returned by a call to the rocalGetBoundingBoxCount
+ */
 extern "C" RocalTensorList ROCAL_API_CALL rocalGetBoundingBoxLabel(RocalContext rocal_context);
+
+/*! \brief get bounding box coordinates
+ * \param [in] rocal_context rocal context
+ * \param [out] buf The user's buffer that will be filled with bounding box coords info for the images in the output batch. It needs to be of size returned by a call to the rocalGetBoundingBoxCords
+ */
```
Comment (swetha097) [inline rocAL/rocAL/include/api/rocal_api_meta_data.h:171]: To add a return

File: rocAL/rocAL/include/api/rocal_api_parameters.h [hunk 15]
```diff
@@ -24,100 +24,116 @@ THE SOFTWARE.
...
+ * \param end end value of the integer range
+ * \return RocalIntParam representing the uniform random integer parameter.
+ */
+extern "C" RocalIntParam ROCAL_API_CALL rocalCreateIntUniformRand(int start, int end);
+
+/*! \brief updates uniform random integer parameter within a specified range.
+ * \param start start value of the integer range
+ * \param end start value of the integer range
+ * \param input_obj  RocalIntParam to be updated.
+ * \return rocal status vcalue
```
Comment (swetha097) [inline rocAL/rocAL/include/api/rocal_api_parameters.h:48]: value

File: rocAL/rocAL/include/api/rocal_api_parameters.h [hunk 16]
```diff
@@ -24,100 +24,116 @@ THE SOFTWARE.
...
+ * \param [in] updating_obj Rocal int Param to update
+ * \return rocal status value
+ */
+extern "C" RocalStatus ROCAL_API_CALL rocalUpdateIntRand(const int *values, const double *frequencies, unsigned size, RocalIntParam updating_obj);
+
+/*! \brief Sets the parameters for a new or existing RocalFloatRandGen object
+ * \param [in] values random float values
+ * \param [in] frequencies frequencies of the values
+ * \param size size of the array
+ * \return random float paraeter
```
Comment (swetha097) [inline rocAL/rocAL/include/api/rocal_api_parameters.h:126]: parameter

File: rocAL/rocAL/source/augmentations/color_augmentations/node_color_twist.cpp [hunk 17]
```diff
@@ -66,4 +65,4 @@ void ColorTwistNode::update_node() {
     _beta.update_array();
     _hue.update_array();
     _sat.update_array();
-}
+}
```
Comment (swetha097) [inline rocAL/rocAL/source/augmentations/color_augmentations/node_color_twist.cpp:68]: To add the EOF back

File: rocAL/rocAL/source/augmentations/node_ssd_random_crop.cpp [hunk 18]
```diff
@@ -20,21 +20,21 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 */
 
-#include <vx_ext_rpp.h>
-#include <graph.h>
 #include "node_ssd_random_crop.h"
+
```
Comment (swetha097) [inline rocAL/rocAL/source/augmentations/node_ssd_random_crop.cpp:24]: To remove any extra lines

File: rocAL/rocAL_pybind/amd/rocal/decoders.out [hunk 19]
```diff
@@ -0,0 +1,392 @@
+# Copyright (c) 2018 - 2023 Advanced Micro Devices, Inc. All rights reserved.
```
Comment (swetha097) [inline rocAL/rocAL_pybind/amd/rocal/decoders.out:1]: What is this file introduced for ?

File: rocAL/rocAL/source/augmentations/effects_augmentations/node_snow.cpp [hunk 20]
```diff
@@ -24,32 +24,29 @@ THE SOFTWARE.
...
-void SnowNode::init( FloatParam *snow_value_param) {
+void SnowNode::init(FloatParam *snow_value_param) {
     _snow_value.set_param(core(snow_value_param));
 }
 
-
 void SnowNode::update_node() {
     _snow_value.update_array();
-}
+}
```
Comment (swetha097) [inline rocAL/rocAL/source/augmentations/effects_augmentations/node_snow.cpp:52]: Add EOF back

**Comment by fiona-gladwin** [comment]
@HazarathKumarM Please resolve the merge conflicts

File: rocAL/rocAL_pybind/amd/rocal/decoders.out (hunk 19 above)
Comment (fiona-gladwin) [reply rocAL/rocAL_pybind/amd/rocal/decoders.out:1]: Please check and remove this, Looks like the extension itself is wrong

File: amd_openvx/openvx/include/vx_ext_amd.h [hunk 21]
```diff
@@ -203,9 +209,7 @@ enum vx_image_attribute_amd_e
     VX_IMAGE_ATTRIBUTE_AMD_OPENCL_BUFFER = VX_ATTRIBUTE_BASE(VX_ID_AMD, VX_TYPE_IMAGE) + 0x01,
     /*! \brief GPU buffer offset. Use a <tt>\ref cl_uint</tt> parameter.*/
     VX_IMAGE_ATTRIBUTE_AMD_GPU_BUFFER_OFFSET = VX_ATTRIBUTE_BASE(VX_ID_AMD, VX_TYPE_IMAGE) + 0x02,
-    /*! \brief Enable user kernel's own GPU buffer for virtual images. Supports only images with
-     * single color plane and stride should match framework's internal alignment. image ROI not supported.
-     * Use a <tt>\ref vx_bool</tt> parameter.*/
+    /*! \brief Enable user kernel's own GPU buffer for virtual images. Supports only images with single color plane and stride should match framework's internal alignment. image ROI not supported. Use a <tt>\ref vx_bool</tt> parameter.*/
```
Comment (fiona-gladwin) [inline amd_openvx/openvx/include/vx_ext_amd.h:212]: If it was changed to single line because of warnings it is fine otherwise revert back

File: rocAL/rocAL_pybind/amd/rocal/decoders.out (hunk 19 above)
Comment (HazarathKumarM) [reply rocAL/rocAL_pybind/amd/rocal/decoders.out:1]: I mistakenly added to the commit. I am removing it now

File: rocAL/rocAL/include/api/rocal_api.h (hunk 2 above)
Comment (HazarathKumarM) [reply rocAL/rocAL/include/api/rocal_api.h:46]: @fiona-gladwin, could you please confirm whether we can remove the commented line?
